django/tasks/backends.py

```diff
@@ -0,0 +1,119 @@
+    def alias(self):
+        return self._backend.alias
+
+    def get_status(self, **kwargs):
+        """
+        Retrieve the status of the task from the backend.
+        """
+        return self._backend.status(self.task_id, **kwargs)
+
+    def get_result(self, **kwargs):
+        """
+        Retrieve the result of the task from the backend if the task has
+        successfully finished or the causing error if the task has failed.
+
+        raises a ResultUnavailable exception when there is no result available:
+            ResultUnknown if status is UNKNOWN
+            ResultPending if status is PENDING
+        """
+        return self._backend.get_result(self.task_id, **kwargs)
+
+
+class BaseBackend(object):
+    def __init__(self, alias, **kwargs):
+        self.alias = alias
+
+    def get_status(self, task_id):
+        """
+        The current status of the task.
+
+        PENDING: the task is currently in-queue, and waiting to be ran.
+        FAILED: the task attempted to run, but failed for some reason
+        SUCCESS: the task successfully ran
+        UNKNOWN: unable to determine the status of the task
+        """
+        raise NotImplementedError
+
+    def get_result(self, task_id):
+        """
+        If the task has ran and returned a result, return that result.
+
+        If not, raise a ResultUnavailable exception:
+            ResultUnknown if status is UNKNOWN
+            ResultPending if status is PENDING
+        """
+        raise NotImplementedError
+
+    def delay(self, task, *args, **kwargs):
+        """
+        Enqueue provided task and return TaskResult instance.
+        """
+        raise NotImplementedError
+
+
+class DummyTaskResult(TaskResult):
+    """
+    TaskResult subclass used by DummyTaskBackend. It embeds the status and the
+    result.
+    """
+    def __init__(self, backend, task_id, status, result):
+        super(DummyTaskResult, self).__init__(backend, task_id)
+        self._status = status
+        self._result = result
+
+    def get_status(self, **kwargs):
+        return self._status
+
+    def get_result(self, **kwargs):
+        return self._result
+
+
+class DummyTaskBackend(BaseBackend):
+    """
+    Dummy backend that executes tasks in-place and returns an instance of
+    DummyTaskResult containing the status (SUCCESS/FAILED) and the result of
+    the task's execution.
+    """
+    def __init__(self, *args, **kwargs):
+        super(DummyTaskBackend, self).__init__(*args, **kwargs)
+        self._next_task_id = 1
+
+    def get_status(self, task_id):
+        return constants.UNKNOWN
+
+    def get_result(self, task_id):
+        raise ResultUnavailable(
+            "DummyTaskBackend doesn't support storing and retrieving results.")
+
+    def delay(self, task, *args, **kwargs):
+        task_id = self._next_task_id
+        self._next_task_id += 1
+        result = None
+        status = constants.SUCCESS
+        try:
+            result = task(*args, **kwargs)
```

Comment:
This code cannot be depended on the ORM - many people use django without the ORM and this would break so many things. Also executing the task after the commit has occurred won't be any different from running it in the same transaction in terms of change visibility - it will still always and reliably see all the changes where a remotely executed task might not.